### PR TITLE
TST: Add assert_array_equal test for big integer arrays.

### PR DESCRIPTION
**Pull request by maxwell-aladago**
Add test for integer arrays containing big numbers.

Related to [#14323](https://github.com/numpy/numpy/issues/14323).

### CODE REVIEW

**Comment by charris** [comment]
Note to self, also need backport to 1.17.

**Comment by mattip** [comment]
LGTM. The original suggestion was to use `np.max` rather than `max`, but I think `max` will work better if `error` is a python object (like in the first case of the added test).

**Comment by eric-wieser** [comment]
`np.max` will be substantially faster than `max` in the regular case, so I think we should keep using it.

**Comment by maxwell-aladago** [comment]
> `np.max` will be substantially faster than `max` for large arrays, so I think we should keep using it.

`np.max` does not work well when the arguments are `Python objects`, I think.

**Comment by eric-wieser** [comment]
> np.max does not work well when the arguments are Python objects, I think.

It should work just fine as long as those object implement `<`.

**Comment by mattip** [comment]
rebased and squashed

**Comment by charris** [comment]
@mattip What is the status of this? Was it passing tests before the rebase?

File: numpy/testing/_private/utils.py [hunk 1]
```diff
@@ -808,12 +810,7 @@ def func_assert_same_pos(x, y, func=isnan, hasval='nan'):
 
                     # note: this definition of relative error matches that one
                     # used by assert_allclose (found in np.isclose)
-                    # Filter values where the divisor would be zero
-                    nonzero = bool_(y != 0)
-                    if all(~nonzero):
-                        max_rel_error = array(inf)
-                    else:
-                        max_rel_error = max(error[nonzero] / abs(y[nonzero]))
+                    max_rel_error = np.max(error / abs(y))
```
Comment (mattip) [inline numpy/testing/_private/utils.py]: I don't understand this change. It will overflow. Perhaps the idea was to use `np.max` in the original code one line 816 and something got messed up in a merge?

Comment (mattip) [reply numpy/testing/_private/utils.py]: reverted this change and added `np.` to `max`, `abs`. Rebased against latest HEAD. Tests seem to pass now.

File: numpy/testing/_private/utils.py [hunk 2]
```diff
@@ -813,7 +814,7 @@ def func_assert_same_pos(x, y, func=isnan, hasval='nan'):
                     if all(~nonzero):
                         max_rel_error = array(inf)
                     else:
-                        max_rel_error = max(error[nonzero] / abs(y[nonzero]))
+                        max_rel_error = np.max(error[nonzero] / np.abs(y[nonzero]))
```
Comment (eric-wieser) [inline numpy/testing/_private/utils.py]: I don't understand why this changes fixes anything - do the tests below fail without it?

Comment (eric-wieser) [reply numpy/testing/_private/utils.py]: I think this whole patch might have become defunct via a rebase

**Comment by mattip** [comment]
The original issue is fixed on master. We may want to add the test, but the code changes are not needed.

File: numpy/testing/_private/utils.py (hunk 2 above)
Comment (mattip) [reply numpy/testing/_private/utils.py]: seems so

**Comment by mattip** [comment]
Test failure is unrelated: window 32 bit cannot install mingw

**Comment by charris** [comment]
At mattip I have this marked for backport to 1.16, 1.17. It looks like I can get away with just copying `assert_array_equal`, as that function has undergone several changes and a reversion in master. Or I could just ignore it. Do we need the tests?

**Comment by mattip** [comment]
@charris I think this test is a nice addition. I rebased and squashed it.

**Comment by charris** [comment]
@mattip OK. I'm still worried about attempting a backport to 1.16-1.17 due to the complicated history of the `utils.py` module. Maybe I'll just take the whole module prior to Pauli's 64 bit mods.

**Comment by mattip** [comment]
`git diff -r v1.16.4...master -- numpy/testing/tests` shows a bunch of changed tests. I don't know how many are relevant

**Comment by charris** [comment]
The failing tests on ppc64le and PyPy  are unrelated, so merging. Thanks @maxwell-aladago @mattip.

**Comment by charris** [comment]
@mattip The current `test_util.py` works fine on 1.16.x after fixing a macro change.
